providers/base/units/dock/jobs.pxu

```diff
@@ -2355,85 +2355,535 @@ _description: Check dock MAC address pass-through function, please make sure fol
+estimated_duration: 10.0
+_summary: USB3 drive removal test
+command: removable_storage_watcher.py -m 500000000 remove usb
+_purpose:
+    This test will check that the system correctly detects the removal of
+    a USB 3.0 storage device
+_steps:
+    1. Click "Test" and remove the USB 3.0 device from the dock.
+       (Note: this test will time-out after 20 seconds.)
+_verification:
+    The verification of this test is automated. Do not change the
+    automatically selected result.
+_siblings:
+    [{"id": "dock/all-reboot-usb3-remove",
+      "depends": "dock/all-reboot-usb3-insert"},
+     {"id": "dock/all-hotplug-usb3-remove",
+      "depends": "dock/all-hotplug-usb3-insert"},
+     {"id": "dock/all-suspend-usb3-remove",
+      "depends": "dock/all-suspend-usb3-insert"},
+     {"id": "dock/all-poweroff-usb3-remove",
+      "depends": "dock/all-poweroff-usb3-insert"}]
+
+id: dock/all-init-audio-speaker-headphone-plug-detection
+category_id: dock-audio
+plugin: user-interact
+estimated_duration: 60.0
+_summary: Headphones recognized when plugged to the dock test
+requires:
+ device.category == 'AUDIO'
+ package.name == 'pulseaudio-utils'
+command: pulse_active_port_change.py sinks
+_purpose:
+    Check that system detects speakers or headphones being plugged in.
+    (Skip this test if the dock does not have headphones connector)
+_steps:
+    1. Prepare a pair of headphones or speakers with a standard 3.5mm jack.
+    2. Locate the speaker / headphone jack on the dock under test.
+    3. Run the test (you have 30 seconds from now on).
+    4. Plug headphones or speakers into the appropriate jack.
+    5. Unplug the device for subsequent tests.
+_verification:
+    Verification is automatic, no action is required.
+    The test times out after 30 seconds (and fails in that case).
+_siblings:
+    [{"id": "dock/all-reboot-audio-speaker-headphone-plug-detection"},
+     {"id": "dock/all-hotplug-audio-speaker-headphone-plug-detection"},
+     {"id": "dock/all-suspend-audio-speaker-headphone-plug-detection"},
+     {"id": "dock/all-poweroff-audio-speaker-headphone-plug-detection"}]
+
+id: dock/all-init-audio-playback-headphones
+category_id: dock-audio
+plugin: user-interact-verify
+flags: also-after-suspend
+estimated_duration: 30.0
+_summary: Headphones output test
+depends: dock/all-init-audio-speaker-headphone-plug-detection
+requires:
+    device.category == 'AUDIO'
+    package.name == 'alsa-base'
+    package.name == 'gir1.2-gst-plugins-base-0.10' or package.name == 'gir1.2-gst-plugins-base-1.0'
+    package.name == 'pulseaudio-utils'
+command:
+    audio_settings.py store --file="$PLAINBOX_SESSION_SHARE"/pulseaudio_settings
+    audio_settings.py set --device=pci --volume=50
+    gst_pipeline_test.py -t 2 'audiotestsrc wave=sine freq=512 ! audioconvert ! audioresample ! autoaudiosink'
+    EXIT_CODE=$?
+    audio_settings.py restore --file="$PLAINBOX_SESSION_SHARE"/pulseaudio_settings
+    exit $EXIT_CODE
```

Comment:
This could be its own Shell script in the `bin` directory. It would be easier for maintenance.